prov/efa/src/efa_mr.c

```diff
@@ -372,6 +399,8 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
 	if (!efa_mr->ibv_mr) {
 		EFA_WARN(FI_LOG_MR, "Unable to register MR: %s\n",
 				fi_strerror(-errno));
+		if (cuda_is_gdrcopy_enabled())
```

Comment:
Don't you also need to check that the handle is set?

Reply:
added an assertion here.